### PR TITLE
BL-1082 APPROVED GH ISSUE: home page Login button still show after the user is logged in

### PR DESCRIPTION
**Pull request by TylrPopcorn**
**BL-1049 APPROVED GH ISSUE: home page Login button still show after the user is logged in**

## Issue
Screen Shot 2022-11-16 at 9 13 19 AM

https://user-images.githubusercontent.com/104395322/202205475-6049bd70-53ab-4dd6-8d63-7cacc01de49d.mov

STEPS TO REPRODUCE:

1. Make sure you are logged OUT and head over to the landing page.
2. Click on the "Login" button and login.
3. After logging in, the homepage should have a button that says "Login" like in the video above.

-- -- --

### SOLUTION:

- Made changes to the LandingPage.js allowing for a fix to the issue currently at hand.
- After logging in, the button should no longer appear in the directory.
- Determined the problem was that there was no verification checking to see if the user is logged in and if anything should be hidden when logged in.

https://user-images.githubusercontent.com/104395322/202209143-15d8bc99-3993-4933-b4c9-601bebea464b.mov

## Type of change

- [🟩] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [🟩] My code follows the style guidelines of this project
- [🟩] I have performed a self-review of my own code
- [🟩] I have removed unnecessary comments/console logs from my code
- [🟩] I have made corresponding changes to the documentation if necessary (optional)
- [🟩] My changes generate no new warnings
- [🟩] I have checked my code and corrected any misspellings
- [🟩] No duplicate code left within changed files
- [🟩] Size of pull request kept to a minimum
- [🟩] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/pages/LandingPage/LandingPage.js [hunk 1]
```diff
@@ -28,12 +28,14 @@ const LandingPage = () => {
...
-              </Button>
-            </div>
+            {!isAuthenticated /** IF the user is logged in, Hide the button */ ? (
+              <div className="button-container-signin">
+                <h3>Already have an account?</h3>
+                <Button type="primary" onClick={() => loginWithRedirect()}>
+                  Log In
+                </Button>
+              </div>
+            ) : null}
```
Comment (joanterm) [inline src/components/pages/LandingPage/LandingPage.js:38]: Great use of a conditional statement! It makes more sense to fix this issue and remove the button once user is logged in.

**Review by joanterm — DISMISSED** [review]

**Review by gpwolfe — COMMENTED** [review]
This does clearly remove the login button from the page after user is logged in. Should we also be removing the mentor/mentee sign-up fields from the page?
I don't know the front end, but the implication of the landing page is 1) Login OR 2) Sign up. If we're removing the one option after login, it seems to make sense to also remove the other, unless people can create accounts without applying to be a mentor or mentee.

File: src/components/pages/LandingPage/LandingPage.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ import FooterLanding from '../FooterLanding/FooterLanding';
 //Images acquired from unsplash.com
 
 const LandingPage = () => {
-  const { loginWithRedirect } = useAuth0();
+  const { loginWithRedirect, isAuthenticated } = useAuth0();
```
Comment (mina-mikhael) [inline src/components/pages/LandingPage/LandingPage.js:19]: relying on auth0 to check whether the user is authenticated to conditionally render the login component is smart

**Review by mina-mikhael — DISMISSED** [review]

**Review by JamesFincher — DISMISSED** [review]
Nice easy fix!

**Comment by TylrPopcorn** [comment]
> This does clearly remove the login button from the page after user is logged in. Should we also be removing the mentor/mentee sign-up fields from the page? I don't know the front end, but the implication of the landing page is 1) Login OR 2) Sign up. If we're removing the one option after login, it seems to make sense to also remove the other, unless people can create accounts without applying to be a mentor or mentee.

This is a good point. I would need clarification on this part and if I should include the rest of the buttons or not.

**Comment by paulstgermain** [comment]
> > This does clearly remove the login button from the page after user is logged in. Should we also be removing the mentor/mentee sign-up fields from the page? I don't know the front end, but the implication of the landing page is 1) Login OR 2) Sign up. If we're removing the one option after login, it seems to make sense to also remove the other, unless people can create accounts without applying to be a mentor or mentee.
> 
> This is a good point. I would need clarification on this part and if I should include the rest of the buttons or not.

@TylrPopcorn I would actually agree - once we've logged in, we have no reason to register. This would be a smooth update to the UI for existing users!

File: src/components/pages/LandingPage/LandingPage.js [hunk 3]
```diff
@@ -28,12 +28,14 @@ const LandingPage = () => {
...
+            {!isAuthenticated /** IF the user is logged in, Hide the button */ ? (
+              <div className="button-container-signin">
+                <h3>Already have an account?</h3>
+                <Button type="primary" onClick={() => loginWithRedirect()}>
+                  Log In
+                </Button>
+              </div>
+            ) : null}
 
             <h3>Ready to Join Underdog Devs?</h3>
```
Comment (paulstgermain) [inline src/components/pages/LandingPage/LandingPage.js:40]: Let's go ahead and include the "Ready to Join" CTA and buttons to this conditional rendering. Once we're logged in, we don't need these anymore!

**Review by paulstgermain — CHANGES_REQUESTED** [review]

**Comment by TylrPopcorn** [comment]
@paulstgermain Ok. I have just made those changes.

**Review by paulstgermain — APPROVED** [review]

**Comment by paulstgermain** [comment]
> @paulstgermain Ok. I have just made those changes.

@TylrPopcorn Perfect, thanks - looking forward to merging this up once we get a few more reviews. Great work! 😄

**Review by gpwolfe — APPROVED** [review]
This looks good from what I can see, wrapping up all application/login in a single disappearing act.

**Review by mina-mikhael — APPROVED** [review]

**Review by JamesFincher — APPROVED** [review]

**Pull request merged**